drivers/storage/portworx/portworx.go

```diff
@@ -153,6 +155,14 @@ func (p *portworx) GetSelectorLabels() map[string]string {
 }
 
 func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster) {
+	if preflight.IsEKS() {
+		toUpdate.Annotations[pxutil.AnnotationIsEKS] = "true"
+		// Do preflight check for eks cloud permission once before px installation
+		if _, ok := toUpdate.Annotations[pxutil.AnnotationPreflightCheck]; !ok {
```

Comment:
[nit] maybe move this so there's a "requiresPreflight" check, set it to true here, and then in a separate if check you can apply the preflight check annotation. That way it scales to multiple different providers easier in the future.